src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -619,7 +682,56 @@ public override SingleNamespaceOrTypeDeclaration VisitEnumDeclaration(EnumDeclar
                 nameLocation: new SourceLocation(node.Identifier),
                 memberNames: memberNames,
                 children: ImmutableArray<SingleTypeDeclaration>.Empty,
-                diagnostics: diagnostics.ToReadOnlyAndFree());
+                diagnostics: diagnostics.ToReadOnlyAndFree(),
+                _nonGlobalAliasedQuickAttributes | quickAttributes);
+        }
+
+        private static QuickAttributes GetQuickAttributes(SyntaxList<AttributeListSyntax> attributeLists)
+        {
+            var result = QuickAttributes.None;
+            foreach (var attributeList in attributeLists)
+            {
+                foreach (var attribute in attributeList.Attributes)
+                    result |= GetQuickAttributes(attribute.Name.GetUnqualifiedName().Identifier.ValueText, inAttribute: true);
+            }
+
+            return result;
+        }
+
+        private static QuickAttributes GetQuickAttributes(string name, bool inAttribute)
+        {
```

Comment:
Let's assert what is the latest known QuickAttribute, as we need to update this when new attributes are added.
Consider even moving this code into an extension that sits next to the `QuickAttributes` definition as they need to be updated together.

Reply:
done.